File: Minimum sum.cpp
class Solution {
  public:
    string minSum(vector<int> &arr) {
        // code here
        sort(arr.begin(), arr.end());
        string sb1, sb2, ans;
        bool flag = true;
        
        // Split digits alternately between sb1 and sb2
        for (int i : arr) {
            if (flag) {
                sb1 += to_string(i);
            } else {
                sb2 += to_string(i);
            }
            flag = !flag;
        }
        
        int i = sb1.size() - 1, j = sb2.size() - 1, carry = 0;
        
        // Add the two strings sb1 and sb2 from right to left
        while (i >= 0 || j >= 0 || carry == 1) {
            int val1 = i >= 0 ? sb1[i] - '0' : 0;
            int val2 = j >= 0 ? sb2[j] - '0' : 0;
            int sum = val1 + val2 + carry;
            carry = sum / 10;
            ans += (sum % 10) + '0';
            i--;
            j--;
        }
        
        // Reverse the result since it was built in reverse order
        reverse(ans.begin(), ans.end());
        
        // Remove leading zeros
        int z = 0;
        while (z < ans.size() && ans[z] == '0') {
            z++;
        }
        
        return ans.substr(z);
    }
};

File: Minimum sum.py
class Solution:
    def minSum(self, arr):
        arr.sort()
        num1 = []
        num2 = []

        for i in range(len(arr)):
            if i % 2 == 0:
                num1.append(str(arr[i]))
            else:
                num2.append(str(arr[i]))

        # Add num1 and num2 as strings without converting to int
        def addStrings(s1, s2):
            i, j = len(s1) - 1, len(s2) - 1
            carry = 0
            result = []

            while i >= 0 or j >= 0 or carry:
                digit1 = int(s1[i]) if i >= 0 else 0
                digit2 = int(s2[j]) if j >= 0 else 0
                total = digit1 + digit2 + carry
                result.append(str(total % 10))
                carry = total // 10
                i -= 1
                j -= 1

            return ''.join(result[::-1]).lstrip('0') or '0'

        s1 = ''.join(num1)
        s2 = ''.join(num2)
        return addStrings(s1, s2)
